docs/javascript-interop.md

```diff
@@ -23,35 +23,49 @@ A Blazor app can invoke JavaScript functions from .NET and .NET methods from Jav
 
 There are times when Blazor .NET code is required to call a JavaScript function. For example, a JavaScript call can expose browser capabilities or functionality from a JavaScript library to the Blazor app.
 
-To call into JavaScript from .NET, use the `IJSRuntime` abstraction, which is accessible from `JSRuntime.Current`. The `InvokeAsync<T>` method on `IJSRuntime` takes an identifier for the JavaScript function you wish to invoke along with any number of JSON serializable arguments. The function identifier is relative to the global scope (`window`). For example if you wish to call `window.someScope.someFunction`, the identifier is `someScope.someFunction`. There's no longer any need to register the function before it's called. The return type `T` must also be JSON serializable.
+To call into JavaScript from .NET, use the `IJSRuntime` abstraction, which is accessible from `JSRuntime.Current`. The `InvokeAsync<T>` method on `IJSRuntime` takes an identifier for the JavaScript function you wish to invoke along with any number of JSON-serializable arguments. The function identifier is relative to the global scope (`window`). If you wish to call `window.someScope.someFunction`, the identifier is `someScope.someFunction`. There's no need to register the function before it's called. The return type `T` must also be JSON serializable.
 
-*exampleJsInterop.js*:
+In the sample app, two JavaScript functions are available to the client-side app that interact with the DOM to receive user input and display a welcome message:
 
-```javascript
-window.exampleJsFunctions = {
-  showPrompt: function (message) {
-    return prompt(message, 'Type anything here');
-  }
-};
-```
+* `showPrompt` &ndash; Produces a prompt to accept user input (the user's name) and returns the name to the caller.
+* `displayWelcome` &ndash; Assigns a welcome message from the caller to a DOM object with an `id` of `welcome`.
 
-*ExampleJsInterop.cs*:
+*wwwroot/exampleJsInterop.js*:
 
-```csharp
-public class ExampleJsInterop
-{
-    public static Task<string> Prompt(string message)
-    {
-        // Implemented in exampleJsInterop.js
-        return JSRuntime.Current.InvokeAsync<string>(
-            "exampleJsFunctions.showPrompt",
-            message);
-    }
-}
-```
+[!code-javascript[](./common/samples/2.x/BlazorSample/wwwroot/exampleJsInterop.js?highlight=2-7)]
+
+Place the `<script>` tag that references the JavaScript file in the *wwwroot/index.html* file:
+
+[!code-html[](./common/samples/2.x/BlazorSample/wwwroot/index.html?highlight=16)]
+
+Don't place a script tag in a component file because the script tag can't be updated dynamically. For more information, see [Add compiler error if there's a &lt;script&gt; element inside a component (aspnet/Blazor &num;552)](https://go.microsoft.com/fwlink/?linkid=872131).
```

Comment:
I'd rather not reference GitHub issues  as part of our docs. All we really need to say here is that you can't use script tags in components. Instead you need to put them in your *index.html* file. If you want to explain more about why that is, we should pull some of Steve's explanation from the GitHub issue into this doc.